src/shogun/labels/BinaryLabels.cpp

```diff
@@ -141,6 +143,32 @@ CLabels* CBinaryLabels::duplicate() const
 
 namespace shogun
 {
+	SG_FORCED_INLINE Some<CBinaryLabels>
+	from_multiclass(CMulticlassLabels* orig)
+	{
+		auto result_vector = orig->get_labels();
+		std::set<int32_t> unique(result_vector.begin(), result_vector.end());
```

Comment:
do we need ordering? :) i guess unordered_set would just do it right?

Reply:
or we dont need the next line, and just take the `.begin()` or?

Reply:
Might be good in order to make sure we have some invariance if the mapping to permutations of the label vector?